### ISSUE TITLE
MAX ON KODI ANDROID TV

### ISSUE THREAD

**Issue by JaJacek65** [matthuisman/slyguy.addons#900]
![20241015_194851](https://github.com/user-attachments/assets/a0432a42-0c81-4220-93c4-f15d91b6b312)

**Comment by matthuisman** [comment]
please provide a kodi debug log.
Also what country are you in and what content did you select to get to that error?

**Comment by JaJacek65** [comment]
country poland. film e.g. Matrix. Log later
-------- Oryginalna wiadomość --------Od: Matt Huisman ***@***.***> Data: 15.10.2024  21:03  (GMT+01:00) Do: "matthuisman/slyguy.addons" ***@***.***> DW: JaJacek65 ***@***.***>, Author ***@***.***> Temat: Re: [matthuisman/slyguy.addons] MAX ON KODI ANDROID TV (Issue #900) 
please provide a kodi debug log.
Also what country are you in and what content did you select to get to that error?

**Comment by RomanJuranek** [comment]
Same error, same version (according to title bar), on Raspberry Pi 3. Movie Lighthouse, region Czech republic.

**Comment by zuzia-dev** [comment]
> please provide a kodi debug log. Also what country are you in and what content did you select to get to that error?

At me the same error on the film “Matrix”, region Poland, system Linux Debian, KODI 22.
[kodi.log.MAX.txt](https://github.com/user-attachments/files/17411571/kodi.log.MAX.txt)

**Comment by zuzia-dev** [comment]
> please provide a kodi debug log. Also what country are you in and what content did you select to get to that error?

I get the same error on “Matrix”, region Poland, Linux Debian, KODI 22, ISA v22.1.7. I add the logs.
[kodi.log.MAX.txt](https://github.com/user-attachments/files/17411571/kodi.log.MAX.txt)

There is a similar problem with the movie “The Zone of Interest.”.
```
2024-10-19 00:02:34.642 T:1378708    info <general>: VideoPlayer::OpenFile: plugin://slyguy.max/?_=play&_play=1&id=6a82c24c-5638-463c-8fa3-529610cf3537
2024-10-19 00:02:34.645 T:1465595   error <general>: ----------------------------------------
2024-10-19 00:02:34.647 T:1465595   error <general>: 
                                                   
2024-10-19 00:02:34.647 T:1465595   error <general>: Exception occurred during processing of request from
2024-10-19 00:02:34.647 T:1465595   error <general>:  
2024-10-19 00:02:34.647 T:1465595   error <general>: ('127.0.0.1', 50188)
2024-10-19 00:02:34.647 T:1465595   error <general>: 
                                                   
2024-10-19 00:02:34.647 T:1465595   error <general>: Traceback (most recent call last):
                                                   
2024-10-19 00:02:34.647 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/usr/lib/python3.11/socketserver.py", line 691, in process_request_thread
                                                       self.finish_request(request, client_address)
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/usr/lib/python3.11/socketserver.py", line 361, in finish_request
                                                       self.RequestHandlerClass(request, client_address, self)
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/home/debian/.kodi/addons/script.module.slyguy/resources/lib/proxy.py", line 128, in __init__
                                                       BaseHTTPRequestHandler.__init__(self, request, client_address, server)
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/usr/lib/python3.11/socketserver.py", line 755, in __init__
                                                       self.handle()
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/usr/lib/python3.11/http/server.py", line 432, in handle
                                                       self.handle_one_request()
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/usr/lib/python3.11/http/server.py", line 420, in handle_one_request
                                                       method()
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/home/debian/.kodi/addons/script.module.slyguy/resources/lib/proxy.py", line 1418, in do_HEAD
                                                       response = self._proxy_request('HEAD', url)
                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>:   File "/home/debian/.kodi/addons/script.module.slyguy/resources/lib/proxy.py", line 1326, in _proxy_request
                                                       raise Exception("File not found: {}".format(real_path))
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>: Exception: File not found: stop.m3u8
                                                   
2024-10-19 00:02:34.648 T:1465595   error <general>: 
2024-10-19 00:02:34.648 T:1465595   error <general>: ----------------------------------------
2024-10-19 00:02:34.648 T:1465595   error <general>: 
                                                   
2024-10-19 00:02:34.757 T:1378708    info <general>: CVideoPlayer::CloseFile()
2024-10-19 00:02:34.757 T:1378708    info <general>: VideoPlayer: waiting for threads to exit
2024-10-19 00:02:35.155 T:1465593    info <general>: Creating InputStream
2024-10-19 00:02:35.664 T:1465593    info <general>: Creating Demuxer
2024-10-19 00:02:35.664 T:1465593   error <general>: OpenDemuxStream - Error creating demuxer
2024-10-19 00:02:35.664 T:1465593    info <general>: CVideoPlayer::OnExit()
2024-10-19 00:02:35.667 T:1378708    info <general>: VideoPlayer: finished waiting
2024-10-19 00:02:35.667 T:1378708    info <general>: CVideoPlayer::CloseFile()
2024-10-19 00:02:35.667 T:1378708    info <general>: VideoPlayer: waiting for threads to exit
2024-10-19 00:02:35.667 T:1378708    info <general>: VideoPlayer: finished waiting
```

**Comment by matthuisman** [comment]
hmm, im unable to reproduce on The Matrix or Zone of interest.
Both hit the same urls as in logs above but no error

Its not that these were actually "Coming soon" movies that have now been added and therefore working?
And the actual issue is that the add-on does not show "Coming soon?

**Comment by matthuisman** [comment]
can anyone confirm "The Matrix" still errors for them?

**Comment by matthuisman** [comment]
ive just pushed [slyguy.max -> v0.1.6](https://github.com/matthuisman/slyguy.addons/commit/27f10675d5d55eb61c876c34b255db69c25f4b06) that should give a better error hopefully when it can't find the edit_id

**Comment by zuzia-dev** [comment]
@matthuisman  
I don't know about the others, but both movies are available in Polish MAX version. Now it is fine. Problem resolved in v0.1.6. Thank you for the quick fix.

**Comment by matthuisman** [comment]
Hmmm, I didn't actually make any changes to fix it. I just added a line to give a better error.

Seems the problem went away itself and that is why I couldn't reproduce I suspect.

Let me know if it comes back and I'll try to investigate it quicker next time 😊
